### ISSUE TITLE
End delimiter logic broken by BEGIN TRANSACTION

### ISSUE THREAD

**Issue by nvoxland** [liquibase/liquibase#3356]
## Description

As reported in https://forum.liquibase.org/t/sql-scripts-with-multiple-go-statement-no-longer-working/7343/6 if you have `BEGIN TRANSACTION` then `COMMIT`, the end delimiters within them are not respected.

## Steps To Reproduce

Create a changeset like:
```
BEGIN TRANSACTION;  
DELETE FROM HumanResources.JobCandidate  
    WHERE JobCandidateID = 13;  
COMMIT;  
```

**Comment by gsolecki** [comment]
Hi, I ran into the same problem, is there any temporary workaround?

**Comment by nvoxland** [comment]
If you set an explicit endDelimeter you can control where the splitting happens. Like endDelimiter="\" or endDelimiter="$$" if there are \'s in the body of your SQL otherwise.

**Comment by bvikhanskiy** [comment]
I use the solution of xvc from here:
https://forum.liquibase.org/t/sql-scripts-with-multiple-go-statement-no-longer-working/7343/6
"**We downgraded to 4.14.0 and everything works.**"

This seems to be the only version that works for me.

**Comment by PasupuletiRohini** [comment]
Hi, we are also blocked by this issue, and we are not able to upgrade to the latest liquibase versions. Is this issue going to be picked up soon?

**Comment by dozsa** [comment]
Seems this was caused by the changes done in https://github.com/liquibase/liquibase/pull/1589

**Comment by MalloD12** [comment]
I've tested this one with the provided changelog (see issue description) and it seems it was fixed some time ago with #4099. As part of #5756 I added a test to validate this issue. 

Thanks,
Daniel.
